impl/src/main/java/org/jboss/weld/util/reflection/Formats.java

```diff
@@ -479,12 +483,38 @@ public static String formatAnnotations(Annotation[] annotations) {
         return formatIterable(annotations, ANNOTATION_LIST_FUNCTION);
     }
 
-    public static String version(Package pkg) {
-        if (pkg == null) {
-            throw new IllegalArgumentException("Package can not be null");
-        } else {
-            return version(pkg.getSpecificationVersion(), pkg.getImplementationVersion());
+    public static String version(Class<?> weldClass) {
+
+        String version = null;
+        String timestamp = null;
+
+        // First try to get weld-build.properties file
+        InputStream in = weldClass.getResourceAsStream(BUILD_PROPERTIES_FILE);
```

Comment:
I was not sure about null-handling semantics of the try-with-resources statement (i.e. what happens if the resource is `null`). But it seems it should be save: https://blogs.oracle.com/darcy/entry/project_coin_null_try_with. I will update the pull.